File: cmsAdminFrontend/src/app/common/service/categoryTemplate.service.ts
import {Injectable} from '@angular/core';
import {HttpHeaders} from '@angular/common/http';
import {Observable} from 'rxjs';
import {ApiService} from '../../shared/services/api.service';
import {BaseCrudService} from '../../shared/services/baseCrud.service';
import {map} from 'rxjs/operators';

@Injectable({
  providedIn: 'root'
})
export class CategoryTemplateService implements BaseCrudService {
  private CATEGORY_TEMPLATE_PATH = '/categoryTemplate';
  private FIND_ALL_BY_ARTICLE_TEMPLATE_NOT_CHOOSE = this.CATEGORY_TEMPLATE_PATH + '/findAllByArticleTemplateNotChoose';

  httpOptions = {
    headers: new HttpHeaders(
      {'Content-Type': 'application/json'}
    )
  };

  constructor( private apiService: ApiService) {

  }

  getAll(): Observable<any> {
    return this.apiService.get(this.CATEGORY_TEMPLATE_PATH).pipe(map(
      response => {
        if (response) {
          return response;
        } else {
          console.log(response);
          return null;
        }
      }
    ));
  }
  checkUrl( url): Observable<any> {
    return this.apiService.post(this.CATEGORY_TEMPLATE_PATH + '/checkUrl', url).pipe(map(
      response => {
        if (response) {
          return response;
        } else {
          console.log(response);
          return null;
        }
      }
    ));
  }
  create( object): Observable<any> {
    return this.apiService.post(this.CATEGORY_TEMPLATE_PATH , object).pipe(map(
      response => {
        if (response) {
          return response;
        } else {
          console.log(response);
          return null;
        }
      }
    ));
  }
  update( object): Observable<any> {
    return this.apiService.put(this.CATEGORY_TEMPLATE_PATH + '/' + object.id , object).pipe(map(
      response => {
        if (response) {
          return response;
        } else {
          console.log(response);
          return null;
        }
      }
    ));
  }

  delete( id): Observable<any> {
    return this.apiService.post(this.CATEGORY_TEMPLATE_PATH + '/delete/' + id).pipe(map(
      response => {
        if (response) {
          return response;
        } else {
          console.log(response);
          return null;
        }
      }
    ));
  }

  findByUrl( url): Observable<any> {
    return this.apiService.get(this.CATEGORY_TEMPLATE_PATH + '/findByUrl/' + url).pipe(map(
      response => {
        if (response) {
          return response;
        } else {
          console.log(response);
          return null;
        }
      }
    ));
  }

  findAllByArticleTemplateNotChoose() {
    return this.apiService.get(this.FIND_ALL_BY_ARTICLE_TEMPLATE_NOT_CHOOSE).pipe(map(
      response => {
        if (response) {
          return response;
        } else {
          console.log(response);
          return null;
        }
      }
    ));
  }
}

File: CmsAdminService/src/main/java/com/demirturk/cms/service/CategoryTemplateService.java
package com.demirturk.cms.service;

import com.demirturk.cms.exception.CmsException;
import com.demirturk.cms.model.dto.CategoryTemplateDto;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface CategoryTemplateService {
    /**
     *
     * @return
     * @throws CmsException
     */
    List<CategoryTemplateDto> getAll() throws CmsException;

    /**
     *
     * @param categoryTemplateDto
     * @return
     * @throws CmsException
     */
    CategoryTemplateDto create(CategoryTemplateDto categoryTemplateDto) throws CmsException;

    /**
     *
     * @param categoryTemplateDto
     * @param id
     * @return
     * @throws CmsException
     */
    CategoryTemplateDto update(CategoryTemplateDto categoryTemplateDto, Long id) throws CmsException;

    /**
     *
     * @param id
     * @return
     * @throws CmsException
     */
    boolean delete(Long id) throws CmsException;

    /**
     *
     * @param url
     * @return
     * @throws CmsException
     */
    boolean checkUrl(String url) throws CmsException;

    /**
     *
     * @return
     * @throws CmsException
     */
    List<CategoryTemplateDto> findAllByArticleTemplateNotChoose() throws CmsException;

    /**
     *
     * @param url
     * @return
     * @throws CmsException
     */
    CategoryTemplateDto findByUrl(String url) throws CmsException;
}
